core/src/main/scala/cats/data/Ior.scala

```diff
@@ -178,9 +178,14 @@ private[data] sealed abstract class IorInstances extends IorInstances0 {
     def show(f: A Ior B): String = f.show
   }
 
-  implicit def catsDataSemigroupForIor[A: Semigroup, B: Semigroup]: Semigroup[Ior[A, B]] = new Semigroup[Ior[A, B]] {
-    def combine(x: Ior[A, B], y: Ior[A, B]) = x.combine(y)
-  }
+  implicit def catsDataMonoidForIor[A: Semigroup, B: Monoid]: Monoid[Ior[A, B]] =
+    new Monoid[Ior[A, B]] {
+      lazy val empty: Ior[A, B] = Ior.right(Monoid[B].empty)
+      def combine(x: Ior[A, B], y: Ior[A, B]) =
+        if (x == empty) y
+        else if (y == empty) x
```

Comment:
I don't feel great about this. It is a weird kind of combine in that `Right(_) combine Left(_)` returns `Both` *unless* you have `Right(Monoid[B].empty)` in which case it returns `Left`.

I think if you want to lift it into a monoid, using `Option[Ior[A, B]]` is the better approach than pressing Right into service.

Reply:
Yep. I felt a bit sketchy while writing this but figured it was worth sharing to get feedback.